apiserver/facades/agent/uniter/networkinfoiaas.go

```diff
@@ -322,7 +344,9 @@ func (n *NetworkInfoIAAS) populateMachineNetworkInfos() error {
 
 	var privateLinkLayerAddress NetInfoAddress
 	for _, addr := range addrs {
-		if addr.SpaceID() == "" {
+		spaceID := addr.SpaceAddr().SpaceID
+
+		if spaceID == "" {
```

Comment:
Can you provide an example where we won't have a space ID? I was under the impression that even on providers without space support everything gets lumped in the alpha space.

Reply:
If we are not doing subnet discovery, the address will not be linked with a space. This and a bunch of other logic in this module can be removed once it happens for all providers. Only VSphere remains AFAIK.